src/java.base/share/classes/java/io/FilterInputStream.java

```diff
@@ -230,7 +230,7 @@ public synchronized void mark(int readlimit) {
      * @see        java.io.FilterInputStream#mark(int)
      */
     @Override
-    public synchronized void reset() throws IOException {
+    public void reset() throws IOException {
```

Comment:
I suspect this change to remove synchronization from `mark()` and `reset()` is intentional. The `FilterInputStream` class doesn't explain the thread safety semantics, nor can I find commit history which would hint on why these (and only these) methods were syncrhonized. But it looks like this synchronization was there to serialize calls to `mark()` and `reset()` on the same `FilterInputStream` from multiple threads. With this synchronization now removed, subclasses or even direct usages of `FilterInputStream` would now be impacted. Potentially outside of the JDK. Would that be a concern?

Reply:
Yes, this is intentional but it would be great to split out this change and get them into the main line in advance. There is a JEP size task required to re-visit all of the locking in java.io. In the case of the input/output streams, most of it is not documented and there are several inconsistencies. In the case of FilterInputStream, it doesn't use synchronization for the other ops.